music21/expressions.py

```diff
@@ -517,21 +519,205 @@ def fillListOfRealizedNotes(
+            self._accid = pitch.makeAccidentalName(accid)
+        self._size: interval.IntervalBase | None = None
         self.quarterLength = 0.125  # 32nd note default
-        self.size = interval.GenericInterval(2)
         self.placement = 'above'
 
-    def realize(self, srcObj: 'music21.note.Note', *, inPlace=False):
+    @property
+    def name(self) -> str:
+        '''
+        returns the name of the Mordent/InvertedMordent, which is generally
+        the class name lowercased, with spaces where a new capital occurs. The
+        name also will include the accidental, if it exists.
+
+        Subclasses can override this as necessary.
+
+        >>> mordent = expressions.Mordent()
+        >>> mordent.name
+        'mordent'
+
+        >>> invertedMordent = expressions.InvertedMordent(accid='sharp')
+        >>> invertedMordent.name
+        'inverted mordent (sharp)'
+
+        '''
+        theName: str = super().name
+        if self.accid is not None:
+            theName += ' (' + self.accid + ')'
+        return theName
+
+    @property
+    def accid(self) -> str | None:
+        return self._accid
+
+    @accid.setter
+    def accid(self, newAccid: str | None):
+        if self._size is not None:
+            raise ExpressionException('Cannot set mordent accid if size already set.')
+
+        # silently ignore invalid accid
+        if newAccid is None:
+            self._accid = None
+        elif pitch.isValidAccidentalName(newAccid):
+            self._accid = pitch.makeAccidentalName(newAccid)
+
+    @property
+    def size(self) -> interval.IntervalBase | None:
+        if self._size is None:
+            raise ExpressionException(
+                'Cannot compute mordent size without srcPitch unless .size has been '
+                + 'explicitly set (use mordent.getSize(srcPitch) instead.)'
+            )
+        return self._size
+
+    @size.setter
+    def size(self, newSize: interval.IntervalBase | None):
+        if self._accid is not None:
+            raise ExpressionException('Cannot set mordent size if accid already set.')
+        self._size = newSize
+
+    @property
+    def direction(self) -> str:
+        return self._direction
+
+    def getSize(
+        self,
+        srcPitch: pitch.Pitch,
+    ) -> interval.IntervalBase:
+        if self._size is not None:
+            return self._size
+
+        if self.direction not in ('up', 'down'):
+            raise ExpressionException('Cannot compute mordent size if I do not know its direction')
+
+        otherPitch: pitch.Pitch = copy.deepcopy(srcPitch)
+        otherPitch.accidental = None
+
+        if self.direction == 'up':
+            otherPitch.transpose(interval.GenericInterval(2), inPlace=True)
+        else:
+            otherPitch.transpose(interval.GenericInterval(-2), inPlace=True)
+
+        if self.accid is not None:
+            otherPitch.accidental = pitch.Accidental(self.accid)
```

Comment:
I'm confused from the comments why the ornament can't store accidental objects?  yes, this would then involve a copy process, but it seems cleaner -- turns have sharp-accidentals attached to them, not 'sharp' strings.